tests/stress_tests/memcheck_tests/tests_pipelines/tests_pipelines.cpp

```diff
@@ -11,130 +11,201 @@
-                  "\" vmpeak=\"" + std::to_string((int) (test_cur_vmpeak * REPORTING_THRESHOLD)) +          \
-                  "\" vmrss=\"" + std::to_string((int) (test_cur_vmrss * REPORTING_THRESHOLD)) +            \
-                  "\" vmhwm=\"" + std::to_string((int) (test_cur_vmhwm * REPORTING_THRESHOLD)) + "\" />");
-
-#define log_info_ref_mem_usage()                                                                \
-        log_info("Reference values of virtual memory consumption:");                            \
-        log_info("VMRSS\t\tVMHWM\t\tVMSIZE\t\tVMPEAK");                                               \
-        log_info(ref_vmrss << "\t\t" << ref_vmhwm << "\t\t" << ref_vmsize << "\t\t" << ref_vmpeak);
-
-#define log_info_cur_mem_usage()                                                                                    \
-        log_info("Current values of virtual memory consumption:");                                                  \
-        log_info("VMRSS\t\tVMHWM\t\tVMSIZE\t\tVMPEAK");                                                                   \
-        log_info(test_cur_vmrss << "\t\t" << test_cur_vmhwm << "\t\t" << test_cur_vmsize << "\t\t" << test_cur_vmpeak);
+
+/**
+ * @brief Class response for encapsulating measure and measurements printing
+ */
+class MemCheckPipeline {
+private:
+    std::array<long, MeasureValueMax> measures;            // current measures
+    std::array<long, MeasureValueMax> start_measures;      // measures before run (will be used as baseline)
+public:
+    MemCheckPipeline() {
+        start_measures[VMRSS] = (long) getVmRSSInKB();
+        start_measures[VMHWM] = start_measures[VMRSS];
+        start_measures[VMSIZE] = (long) getVmSizeInKB();
+        start_measures[VMPEAK] = start_measures[VMSIZE];
+        start_measures[THREADS] = (long) getThreadsNum();
+
+        std::copy(start_measures.begin(), start_measures.end(), measures.begin());
+    }
+
+    ~MemCheckPipeline() {
+        // message required for DB data upload
+        log_info("Current values of virtual memory consumption:");
+        print_measures();
+    }
+
+    void do_measures() {
+        measures[VMRSS] = (long) getVmRSSInKB();
+        measures[VMHWM] = (long) getVmHWMInKB();
+        measures[VMSIZE] = (long) getVmSizeInKB();
+        measures[VMPEAK] = (long) getVmPeakInKB();
+        measures[THREADS] = (long) getThreadsNum();     // TODO: resolve *-32295
+    }
+
+    /**
+     * @brief Returns measurements aligned on a baseline
+     */
+    std::array<long, MeasureValueMax> get_measures() {
+        std::array<long, MeasureValueMax> temp;
+        std::transform(std::begin(measures), std::end(measures), std::begin(start_measures), std::begin(temp),
+                       [](long measure, long start_measure) -> long {
+                           return measure - start_measure;
+                       });
+
+        return temp;
+    }
+
+    std::string get_measures_as_str() {
```

Comment:
Please document public API of MemCheckPipeline

Reply:
Done